torch/_refs/__init__.py

```diff
@@ -5580,7 +5583,7 @@ def full_like(
         requires_grad=requires_grad,
         memory_format=memory_format,
     )
-    return fill(e, fill_value)
+    return prims.fill(e, fill_value).to(dtype)
```

Comment:
Is `.to(dtype)` needed here? That looks like a workaround for a problem lurking somewhere inside. `e` should have the `dtype` as passed, if passed, otherwise `a.dtype`, and fill should not change it.

Reply:
It's not. I'll remove it